File: src/main/java/tn/esprit/tpfoyer/entity/Etudiant.java
package tn.esprit.tpfoyer.entity;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;
import lombok.extern.slf4j.Slf4j;

import java.io.Serializable;
import java.util.Date;
import java.util.Set;

@Entity
@Getter
@Setter
@AllArgsConstructor
@ToString
@NoArgsConstructor
public class Etudiant implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idEtudiant;
    private String NomEt;
    private String prenomEt;
    private Long cin;
    private String ecole;
    private Date dateNaissance;

}

File: src/app/models/Etudiant.ts
import { Reservation } from './Reservation';

export class Etudiant {
    idEtudiant: number;
    nomEt: string;
    prenomEt: string;
    cin: number;
    ecole: string;
    dateNaissance: Date;
    reservations?: Reservation[];

    constructor(
        idEtudiant: number,
        nomEt: string,
        prenomEt: string,
        cin: number,
        ecole: string,
        dateNaissance: Date,
        reservations?: Reservation[]
    ) {
        this.idEtudiant = idEtudiant;
        this.nomEt = nomEt;
        this.prenomEt = prenomEt;
        this.cin = cin;
        this.ecole = ecole;
        this.dateNaissance = dateNaissance;
        this.reservations = reservations;
    }
}
